core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -447,6 +457,32 @@ extern "C" fn new_confirm_action(n_args: usize, args: *const Obj, kwargs: *mut M
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_confirm_single(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let begin: StrBuffer = kwargs.get_or(Qstr::MP_QSTR_begin, StrBuffer::empty())?;
+        let middle: StrBuffer = kwargs.get_or(Qstr::MP_QSTR_middle, StrBuffer::empty())?;
+        let end: StrBuffer = kwargs.get_or(Qstr::MP_QSTR_end, StrBuffer::empty())?;
+
+        let ops = OpTextLayout::new(theme::TEXT_NORMAL)
+            .text_normal(begin)
```

Comment:
This configuration of normal-demibold-normal works for this one use-case, but cannot we make it more general - so that python can supply e.g. `list[tuple[str, int]]` with strings and font IDs?

We are already supplying and processing dynamic-sized lists in `confirm_properties`.

(One benefit would be also not having to define `MP_QSTR_{begin,middle,end}`)

Some quick pseudocode:
```rs
let items: Obj = kwargs.get(Qstr::MP_QSTR_items)?;
let mut iter_buf = IterBuf::new();
let iter = Iter::try_from_obj_with_buf(items, &mut iter_buf)?;
let mut ops = OpTextLayout::new(theme::TEXT_NORMAL);
for item in iter {
    let [string, font]: [Obj; 2] = iter_into_array(item)?;
    let string: StrBuffer = string.try_into()?;
    let font: i16 = font.try_into()?;
    match font {
        Font::NORMAL => {
            ops = ops.text_normal(string);
        },
        Font::BOLD => {
            ops = ops.text_bold(string);
        }
        ...
    }
}
```

Reply:
Good idea, don't particularly enjoy `MP_QSTR_{begin,middle,end}` either. But I don't think we should expose the font choice to python - there's a precedent in `confirm_with_info`/`confirm_more` but I don't think we should add more as this goes against #2634.

In 03c7bfe293bab0ad5a499f7095846dae5cd87e6e I implemented this but with bool instead of font number, indicating whether that part of the text should be emphasized (by using demibold font on TT). This should work across hw models.

Reply:
Good compromise, I agree that sending the `font ID` is unnecessary detail for `micropython`, sending just `is_emphasized` leaves it all to `Rust` to decide.